code/modules/mob/living/simple_animal/hostile/goose.dm

```diff
@@ -25,33 +27,144 @@
 	faction = list("neutral")
 	attack_same = TRUE
 	gold_core_spawnable = HOSTILE_SPAWN
-	var/icon_resting = "goose_sit"
+	var/random_retaliate = TRUE
+	var/icon_vomit_start = "vomit_start"
+	var/icon_vomit = "vomit"
+	var/icon_vomit_end = "vomit_end"
+
+/mob/living/simple_animal/hostile/retaliate/goose/handle_automated_movement()
+	. = ..()
+	if(prob(5) && random_retaliate == TRUE)
+		Retaliate()
 
+/mob/living/simple_animal/hostile/retaliate/goose/vomit
+	name = "Birdboat"
+	real_name = "Birdboat"
+	desc = "It's a sick-looking goose, probably ate too much maintenance trash. Best not to move it around too much."
+	gender = MALE
+	response_help  = "pets"
+	response_disarm = "gently pushes aside"
+	response_harm   = "kicks"
+	gold_core_spawnable = NO_SPAWN
+	random_retaliate = FALSE
+	var/vomiting = FALSE
+	var/vomitCoefficient = 1
+	var/vomitTimeBonus = 0
+	var/datum/action/cooldown/vomit/goosevomit
 
-/mob/living/simple_animal/hostile/retaliate/goose/toggle_ai(togglestatus)
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/Initialize()
 	. = ..()
-	if(!key)
-		if(AIStatus != AI_ON)
-			set_resting(TRUE)
-		else
-			set_resting(FALSE)
+	goosevomit = new
+	goosevomit.Grant(src)
+
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/Destroy()
+	QDEL_NULL(goosevomit)
+	return ..()
 
-/mob/living/simple_animal/hostile/retaliate/goose/update_resting()
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/examine(user)
 	. = ..()
-	if(resting)
-		wander = FALSE
-	else
-		wander = TRUE
-	update_icon()
+	. += "<span class='notice'>Somehow, it still looks hungry.</span>"
 
-/mob/living/simple_animal/hostile/retaliate/goose/handle_automated_movement()
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/attacked_by(obj/item/O, mob/user)
 	. = ..()
-	if(prob(5))
-		Retaliate()
+	feed(O)
 
-/mob/living/simple_animal/hostile/retaliate/goose/proc/update_icon()
-	if(stat != DEAD)
-		icon_state = resting ? icon_resting : icon_living
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/proc/feed(obj/item/O)
+	var/obj/item/reagent_containers/food/tasty = O
+	if(!istype(O, /obj/item/reagent_containers/food))
```

Comment:
When you're typechecking a var for the same type it's cast as you can shorthand it to
```suggestion
	if(!istype(O))
```